protocol/messenger.go

```diff
@@ -3833,6 +3833,40 @@ func (m *Messenger) handleRetrievedMessages(chatWithMessages map[transport.Filte
 							continue
 						}
 
+					case protobuf.SyncActivityCenterNotifications:
```

Comment:
[Light suggestion] This `handleRetrieveMessages` function has more than 1000 lines of code already :cry: :godmode: It would be ideal if we stop growing it like a god function IMO.

Reply:
A blasphemy has been spoken! The ways of Yog-Sothoth are unknowable, beyond even the merest glimpse.

Also, think of this function as a register, we can do some basic refactoring of this function, but fundamentally we will still need to pass a protobuf into something and then make a decision about what to do with that protobuf.

Reply:
hauhauahau, I think I understand what you mean @Samyoul. But still, this function has too many responsibilities, and it has too many reasons to change. Each case could be handled by individual functions, easily testable in isolation if we so wanted. It would be a win to me if we could make `handleRetrievedMessages` as dumb as possible, so that it essentially becomes a long list of match by type + dispatcher.

A simple extract refactor for the 2 cases added in this PR would be something I would personally do, to start to break free from the holy pattern! But other ideas could follow in the future, so more blasphemy can be spoken.

Some could say, don't change a winning team, so perhaps `handleRetrievedMessages` is such a team ;)

Reply:
Yes @ilmotta, I agree with you and I have floated the idea of refactoring this function in particular in the past.

- "make the `handleRetrievedMessages` as dumb as possible" - Winning ✅ 
- "Each case could be handled by individual functions" - Winning ✅ 
- "easily testable in isolation" - Winning ✅ 
- "more blasphemy" - [EPIC WINNING!! ✅ ](https://www.youtube.com/watch?v=9QS0q3mGPGg)

Reply:
Hahaha, actually, when I trying to create my first PR relate to `handleRetrievedMessages`, i saw this issue, it's too big! I was trying pick up reflection of tutorial to see what i can do 😅

Reply:
We are on the same page...

<img src="https://github.com/status-im/status-go/assets/46027/83ffb198-97c6-41b8-909a-ff16098750cd" height="400" />

Reply:
Although I would say that the difficulty(?) with breaking each case into a dedicated function is that each handler is basically the testable unique kernel. But the rest of the functionality is varying forms of pre and post handling, some cases use identical set ups, others use a combination of common elements and other have completely custom handling.

I'd be very interested in seeing the solution / proposals.